include/openthread/error.h

```diff
@@ -239,6 +239,11 @@ typedef enum OT_MUST_USE_RESULT otError
      */
     OT_ERROR_REJECTED = 37,
 
+    /**
+     * A transmission could not take place due to coex grant timeout.
+     */
+    OT_ERROR_COEX_GRANT_TIMEOUT = 38,
+
```

Comment:
Can consider using `OT_ERROR_ABORT` to indicate this? 

I think `ABORT` error was intended to indicate situation like this and I think radio platform should not return `CHANNEL_ACCESS_FAILTURE` for this (there are different behaviors on CCA failure).

Reply:
Thanks! Make sense to me. I think we can use `ABORT` for now.

Just one concern: I think `ABORT` is kind of general and some platforms may use this error for some platform-specific errors in radio. On nrf528xx, it's not used and we can use it for coex grant timeout. If some other platform already uses `ABORT` for some cases and we want to know which cases are exactly caused by coex, then we cannot differentiate them. If someday we really need to add an error code, can we do that?

Reply:
@abtink

After more thinking on this, for future consideration I think it's necessary to have a specialized error for coex issue so that OT is aware of what has happened exactly. If we use `ABORT` for coex error and specify that if a platform gets coex error it should return `ABORT`, then on some platforms, `ABORT` errors may contain some other cases. 

To add an error code, we document it and carefully update all the places that may handle this new error. For host-RCP compatibility issues, let's use RCP_API_VERSION to ensure the host supports the new error. Thoughts?

Reply:
> I think it's necessary to have a specialized error for coex issue so that OT is aware of what has happened exactly

- What other situations radio platform can return `ABORT`? I am not sure there are many cases `ABORT` will be used.
- Why OT needs to be aware of it? Do we need to take any specific action based on coex error on OT core code?
-  If the goal is to collect info/stats, perhaps we can create another mechanism t o report this?

>  For host-RCP compatibility issues, let's use RCP_API_VERSION to ensure the host supports the new error.

Are you suggesting we break backward compatibility. I don't think it is a good  idea.

The challenging part is if RCP is newer version and host is older. RCP has no way of knowing that host is older to change its behavior and not emit this new error. Host being older will not know how to handle this error and will assert. 

Regarding use of `RCP_API_VERSION`, we decided to ensure backward compatibility unless there is strong reason/justification and no other solution (cc: @zhanglongxi).

Reply:
> What other situations radio platform can return ABORT? I am not sure there are many cases ABORT will be used.

Currently on nRF platform, in context of TX, `ABORT` is not used at all. So it's totally fine to use `ABORT` with nRF platform.

What I think is, can we specify in `radio.h` that `ABORT` exactly means coex error in the context of TX so that other platforms won't use `ABORT` for other cases in their radio platform implementation? 

> Why OT needs to be aware of it? Do we need to take any specific action based on coex error on OT core code?
If the goal is to collect info/stats, perhaps we can create another mechanism to report this?

Yes exactly, the goal is to collect some stats. I think we can create some other mechanism but it would be preferrable if we can count them with the same way we count other tx errors and frame numbers. For example, if we have another flow to count this specific error and report it, there will be more chances it being inconsistent with the total tx frame counts.

Regarding the compatibility, I totally agree. Let's ensure backward compatibility as possible.

---

What about we specify `ABORT` as coex issues in our radio platform API header? Currently `OT_ERROR_ABORT` is described as `OT_ERROR_ABORT when transmission was aborted for other reasons.` which is kind of general. However with such change, we'd better add the error code (OT_ERROR_OTHER) for the original ABORT error. Then we still need to update the handling on host and get compatibility issue. @abtink

Reply:
Here is my suggestion:
- We leave definition of `ABORT` error in `radio.h` as is and generic.
- We provide counter info tracking number of `ABORT` errors (already do as `mTxErrAbort` counter).
- It can be up to each platform/project to determine what `ABORT` means and how to interpret the counter

I don't think it is a good idea to say `ABORT` is only for coex:
- Coex only makes sense for platforms where there are other radios and therefore coex mechanism is helpful and used
- On a platform with single 15.4 radio, `ABORT` error can still be used for other purposes

Reply:
Make sense to me. Thanks!